src/controllers/auth/signup.ts

```diff
@@ -0,0 +1,21 @@
+import * as Interfaces from "../../interfaces/index";
+import * as Utils from "../../utils/index";
+
+const signUp: Interfaces.Controllers.Async = async (req, res, next) => {
+  const data: Interfaces.User = req.body;
+  try {
+    const user = await Utils.Auth.db.createUser(data);
+
+    return res.json(
+      Utils.Response.success({
+        id: user.id,
+        email: user.email,
+        password: user.password,
+      })
+    );
+  } catch (err) {
+    return next(err);
+  }
```

Comment:
Won't you check if user already exist?

Reply:
This is getting verified in the custom signup validator middleware. So , I have not put any checks in here as the custom validator would anyways throw error

Reply:
Okay got it